File: src/test/rustdoc-js/summaries.js
// ignore-tidy-linelength

const QUERY = ['summaries', 'summaries::Sidebar', 'summaries::Sidebar2'];

const EXPECTED = [
    {
        'others': [
           { 'path': '', 'name': 'summaries', 'desc': 'This <em>summary</em> has a link and <code>code</code>.' },
        ],
    },
    {
        'others': [
            { 'path': 'summaries', 'name': 'Sidebar', 'desc': 'This <code>code</code> will be rendered in a code tag.' },
        ],
    },
    {
        'others': [
            { 'path': 'summaries', 'name': 'Sidebar2', 'desc': '' },
        ],
    },
];

File: src/test/rustdoc-js/summaries.rs
#![crate_type = "lib"]
#![crate_name = "summaries"]

//! This *summary* has a [link] and `code`.
//!
//! This is the second paragraph.
//!
//! [link]: https://example.com

/// This `code` will be rendered in a code tag.
///
/// This text should not be rendered.
pub struct Sidebar;

/// ```text
/// this block should not be rendered
/// ```
pub struct Sidebar2;
